Jenkinsfile

```diff
@@ -0,0 +1,306 @@
+                  env.BULK_UPLOAD_FILE = "apm-agent-bulk-${env.NOW_ISO_8601}.json"
+                }
+                sh """#!/bin/bash
+                ./scripts/jenkins/run-benchmarks.sh
+                """
+              }
+            }
+          } 
+          post {
+            always {
+              sendBenchmarks(file: "${BASE_DIR}/${BULK_UPLOAD_FILE}", index: "benchmark-java", bulk: true)
+              archiveArtifacts(allowEmptyArchive: true, 
+                artifacts: "${BASE_DIR}/${RESULT_FILE},${BASE_DIR}/${BULK_UPLOAD_FILE}", 
+                onlyIfSuccessful: false)
+            }
+          }
+        }
+        /**
+         run Go integration test with the commit version on master branch.
+        */
+        stage('Integration test master') { 
+          agent { label 'linux && immutable' }
+          when { 
+            beforeAgent true
+            allOf { 
+              //branch 'master';
+              environment name: 'integration_test_master_ci', value: 'true' 
+            }
+          }
+          steps {
+            build(
+              job: 'apm-server-ci/apm-integration-test-axis-pipeline', 
+              parameters: [
+                string(name: 'BUILD_DESCRIPTION', value: "${BUILD_TAG}-INTEST"),
+                booleanParam(name: "go_Test", value: false),
+                booleanParam(name: "java_Test", value: true),
+                booleanParam(name: "ruby_Test", value: false),
+                booleanParam(name: "python_Test", value: false),
+                booleanParam(name: "nodejs_Test", value: false)],
+              wait: true,
+              propagate: true)
+          }
+        }
+        
+        /**
+         run Go integration test with the commit version on a PR.
+        */
+        stage('Integration test PR') { 
+          agent { label 'linux && immutable' }
+          when { 
+            beforeAgent true
+            allOf { 
+              changeRequest()
+              environment name: 'integration_test_pr_ci', value: 'true' 
+            }
+          }
+          steps {
+            build(
+              job: 'apm-server-ci/apm-integration-test-pipeline', 
+              parameters: [
+                string(name: 'BUILD_DESCRIPTION', value: "${BUILD_TAG}-INTEST"),
+                string(name: 'APM_AGENT_JAVA_PKG', value: "${BUILD_TAG}"),
+                booleanParam(name: "go_Test", value: false),
+                booleanParam(name: "java_Test", value: true),
+                booleanParam(name: "ruby_Test", value: false),
+                booleanParam(name: "python_Test", value: false),
+                booleanParam(name: "nodejs_Test", value: false),
+                booleanParam(name: "kibana_Test", value: false),
+                booleanParam(name: "server_Test", value: false)],
+              wait: true,
+              propagate: true)
+          }
+        }
+      }
+    }
+        
+    stage('Documentation') { 
```

Comment:
cool, finally a docs build :)

Reply:
yep, if you see something that I missed or it is needed, please mention it. Also I put the steps to run on master branch, a PR, or both by my own criteria tell me if it is correct.

checkout - all
build - all
test - all
benchmarks - branches (master, 6.x, ...) not in PRs
full integration test - master (disabled for the moment)
short integration test - PRs and other branches that are not master (disabled for the moment)
documentation - master

Reply:
What do you refer to when you say integration tests? Do you mean testing this (https://github.com/elastic/apm-integration-testing/tree/master/docker/java/spring/testapp) app against a real APM Server and ES?

The Java agent also has some integration tests or maybe you could call them smoke tests for different servers and databases. They are executed on the `verify` goal. `test` only executes the unit tests.

What is the difference between full integration test and short integration test?

Reply:
>What do you refer to when you say integration tests? Do you mean testing this (https://github.com/elastic/apm-integration-testing/tree/master/docker/java/spring/testapp) app against a real APM Server and ES?

yep, https://github.com/elastic/apm-integration-testing/blob/master/scripts/ci/java.sh

>The Java agent also has some integration tests or maybe you could call them smoke tests for different servers and databases. 

smoke test is fine

>They are executed on the verify goal. test only executes the unit tests.

Noted

Reply:
>What is the difference between full integration test and short integration test?

The short  integration test will only test the PR against the stable version of APM Server + Elastic Stack, the other will test against a matrix of versions, this is made on other agents ensure that you version will work with different combinations of APM Server + Elastic Stack. TBD which matrix versions we want to use.